### ISSUE TITLE
`isStandardSyntaxAtRule` calls an at-rule spelled without a space in front of its options a call to a detached ruleset

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#357]
`isStandardSyntaxAtRule` reads an at-rule spelled without a space in front of its options as a call to a detached ruleset, and three rules then pass it over.

The util turns a Less detached-ruleset call away by its shape — no block, an empty `raws.afterName`, and parameters opening with a parenthesis:

```js
if ((`variable` in atRule && atRule.variable) || (!atRule.nodes && atRule.raws.afterName === `` && atRule.params[0] === `(`)) return false
```

An ordinary at-rule spelled without a space in front of its options has exactly that shape. `postcss-less` hands all of these over with `afterName: ""` and parameters opening with a parenthesis:

```less
a { @import(reference) "x"; }
a { @supports(a: b); }
a { @layer(l); }
a { @plugin(args) "p"; }
a { @whatever(x); }
```

Every one of them is a real at-rule, and `less@4.9.0` compiles all five. What the util says of them is that they are not standard syntax, so every rule reading it walks past.

## What that costs today

`at-rule-name-case` under `lower` fixes one spelling of an at-rule and passes over the other:

```less
a { @IMPORT (reference) "x"; }   /* --fix writes @import */
a { @IMPORT(reference) "x"; }    /* --fix writes nothing, and nothing is reported */
```

`at-rule-semicolon-space-before` under `always` reports the first and says nothing about the second. `at-rule-semicolon-newline-after` and `atRuleNameSpaceChecker` read the same util and are reachable the same way; `no-extra-semicolons` reads it too.

## Why no oracle sees it

All five oracles ask about the output of a fix, and these are false negatives: nothing is written, so nothing converges wrongly, no comment is lost and no node disappears. No fixture in the corpus spells an at-rule without that space either.

## What would end it

A call to a detached ruleset takes no arguments, so its parameters are `()` and nothing else — measured against Less, `@dr()` is a call and `@dr ()` is an at-rule it asks for a semicolon behind. Reading `params === "()"` rather than `params[0] === "("` tells the two apart. `requiresTrailingSemicolon`, added for [#309](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/309), carries that reading already and does not call this util for exactly this reason.

Two other shapes are worth measuring in the same pass, since both are Less constructs this util names by one spelling of several: a variable declaration is marked `variable` only where the colon lands in `raws.afterName`, so `@v : pink` and `@v\t: pink` are called standard at-rules, while `@v:pink` is turned away by the `@content` branch named next rather than by anything the util knows about Less; and `!atRule.nodes && atRule.params === ""`, written for the Sass `@content`, also catches a Less mixin call spelled without parentheses, which the `mixin` mark answers for anyway.

Found while writing the branch for #309, which is where the five spellings above were measured.
